zkfacade/src/main/java/com/yahoo/vespa/curator/SingletonManager.java

```diff
@@ -302,9 +306,20 @@ private void renewLease() {
             }
         }
 
+        private void cleanOrphans() {
+            List<String> orphans = null;
+            try {
+                for (String orphan : orphans = curator.framework().getZookeeperClient().getZooKeeper().getEphemerals(path.getAbsolute()))
+                    curator.delete(path.append(orphan));
```

Comment:
Before you ask:

1. When a new singleton is registered, potentially as a result of a reload of user components (only), it will be registered with this same singleton manager, for which there should only ever be one lock.
2. When a new singleton manager is created in the same container, as a result of curator or zookeeper config changing, the new curator will use a different client, and see a different set of ephemeral nodes.

Reply:
Leave a comment on getEphemerals()?  It only fetches the ephemerals owned by this session, which is not obvious but important.